Tasks/Kubernetes/Tests/L0.ts

```diff
@@ -0,0 +1,390 @@
+        console.log(tr.stderr);
+        done();
+    });
+
+    it('Run fails when the user provides a wrong location for kubectl', (done:MochaDone) => {
+        let tp = path.join(__dirname, 'TestSetup.js');
+        let tr : ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+        process.env[shared.TestEnvVars.command] = shared.Commands.get;
+        process.env[shared.TestEnvVars.arguments] = "pods";    
+        process.env[shared.TestEnvVars.versionOrLocation] = "location";
+        process.env[shared.TestEnvVars.specifyLocation] = shared.formatPath("wrongDir/kubectl.exe");
+        process.env[shared.isKubectlPresentOnMachine] = "false";
+        tr.run();
+        
+        assert(tr.failed, 'task should have failed');
+        assert(tr.invokedToolCount == 0, 'should have invoked tool 0 times. actual: ' + tr.invokedToolCount);
+        assert(tr.stderr.length > 0 || tr.errorIssues.length, 'should have written to stderr');
+        assert(tr.stdout.indexOf(`Not found ${shared.formatPath("wrongDir/kubectl.exe")}`) != -1, "kubectl get should not run");
+        console.log(tr.stderr);
+        done();
+    });
+
+    it('Runs successfully for kubectl apply using configuration file', (done:MochaDone) => {
+        let tp = path.join(__dirname, 'TestSetup.js');
+        let tr : ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+        process.env[shared.TestEnvVars.command] = shared.Commands.apply;
+        process.env[shared.TestEnvVars.useConfigurationFile] = "true";    
+
+        tr.run();
+        assert(tr.invokedToolCount == 1, 'should have invoked tool one times. actual: ' + tr.invokedToolCount);
+        assert(tr.stderr.length == 0 || tr.errorIssues.length, 'should not have written to stderr');
+        assert(tr.succeeded, 'task should have succeeded');
+        assert(tr.stdout.indexOf(`[command]kubectl --kubeconfig ${shared.formatPath("newUserDir/config")} apply -f ${shared.formatPath("dir/deployment.yaml")}`) != -1, "kubectl apply should run");
+        console.log(tr.stderr);
+        done();
+    });
+
+    it('Runs successfully for kubectl delete', (done:MochaDone) => {
+        let tp = path.join(__dirname, 'TestSetup.js');
+        let tr : ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+        process.env[shared.TestEnvVars.command] = shared.Commands.delete;
+        process.env[shared.TestEnvVars.arguments] = "pod -all";
+
+        tr.run();
+
+        assert(tr.invokedToolCount == 1, 'should have invoked tool one times. actual: ' + tr.invokedToolCount);
+        assert(tr.stderr.length == 0 || tr.errorIssues.length, 'should not have written to stderr');
+        assert(tr.succeeded, 'task should have succeeded');
+        assert(tr.stdout.indexOf(`[command]kubectl --kubeconfig ${shared.formatPath("newUserDir/config")} delete pod -all`) != -1, "kubectl delete should run");
+        console.log(tr.stderr);
+        done();
+    });
+
+    it('Runs successfully for kubectl exec', (done:MochaDone) => {
```

Comment:
We dont have to add test cases for each command.

Reply:
Removed duplicate test cases